### PR TITLE
Added logger commands

### PR DESCRIPTION
**Pull request by jcberquist**

### CODE REVIEW

File: logger/delete.cfc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			
+		if( !toDetails.path.len() ) {
+			error( "The location for the server couldn't be determined.  Please check your spelling." );
+		}
+		
+		// Read existing config
+		var oConfig = CFConfigService.determineProvider( toDetails.format, toDetails.version )
+			.read( toDetails.path );
+
+		// Get the caches and remove the requested one
```
Comment (bdw429s) [inline logger/delete.cfc]: Make sure you find/replace all instance of "caches".

**Comment by bdw429s** [comment]
@jcberquist I see this says work in progress.  Let me know when it's ready.  For the most part if looks good.  I do the same thing it looks like you did, which is copy another set of commands and just find/replace as necessary.  It's mostly boilerplate.

**Comment by jcberquist** [comment]
This is a WIP - the delete doesn't actually do anything. I wanted to check with you on that. I noticed that the `writeDatasources()` method in `BaseLucee.cfc` for instance has the line 
```javascript
datasources.XMLChildren = [];
```
This means that it erases all datasources in the xml config file that aren't in the imported cfconfig data. The check on [L488-L495](https://github.com/Ortus-Solutions/cfconfig/blob/master/models/BaseLucee.cfc#L488-L495) will never return as true, since the xmlChildren have already been cleared.

So I am not sure what the desired behavior is here. In order for the delete to work I could reproduce the `datasources.XMLChildren = []` approach, but the following check in the code made me wonder if that was actually desired. Should I be looking for a different approach to deleting?

File: logger/delete.cfc (hunk 1 above)
Comment (jcberquist) [reply logger/delete.cfc]: 👍

**Comment by bdw429s** [comment]
@jcberquist Yes at one point I had CFConfig never removing old records, but then I found I couldn't delete anything.  Right now, save always removes everything and then adds back what is in object.  My goal is to add some sort of global "append" flag that would allow you to import a JSON file and only add new stuff.  We'd also need to figure out how to make delete for a specific item.  What's tricky is that the flow goes like this:

1. Config is loaded from somewhere into the CFC instance
2. Data in CFC is manipulated (added/deleted)
3. CFC is asked to write itself back out

So we need something we can pass to the global write method that can set some internal flag that tells the CFC if it's wiping all data that doesn't currently exist in the CFC instance, or if it's only adding new stuff in.  I don't think that's very hard, it was just annoying enough at the time that I punted on it and just did a full overwrite.  But that's why you see some code in place that assumes there might be existing data there already.  Feel free to help me tackle the "append" sort of functionality in a separate pull.  It doesn't really apply to the `cfcconfig XXX save` or `cfconfig XXX delete` commands (which is partially why I punted on it), but will be necessary if we ever want to import a series of JSON files and have the total datasources/caches/mappings/etc be cumulative.

**Comment by jcberquist** [comment]
I really need to look into that post you had about symlinking into the modules directory :)

I will make another upstream pull request to clear out the current loggers data before writing out. With that in place, these commands should all work now.

**Comment by bdw429s** [comment]
> I really need to look into that post you had about symlinking into the modules directory :)
If you have the 3.9.0 release candidate just run `link` from the root of your module.  Note these caveats:

1. You will want to have run `install` inside the module to gather its dependencies first
2. You'll want to  uninstall the ForgeBox version of the module first so they don't collide.  
3. You'll probably need to separately link `commandbox-cfconfig` and `cfconfig` and make sure you _don't_  run `install` in `commandbox-cfconfig`.  It gets a little tricky once you're dealing with several modules that all depend on each other, but you want to make sure there are more than one version of a given model present at the same time because CommandBox will just randomly pick one of them to load which might not be helpful.

**Comment by bdw429s** [comment]
Hi @jcberquist , I forgot where we are on this on.  Are you still adding anything or is this ready to merge?

**Comment by jcberquist** [comment]
Hey @bdw429s, this should be good to go now - spent some time getting set up as described above and things seem to be working, thanks!

**Pull request merged**